File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnPlaylist.cs

namespace GracenoteSDK {

/**
* <b>Experimental</b>: Provides access to information regarding the underlying
* GNSDK playlist library. For working with user collections to create playlists
* see GnPlaylistCollection.
*/
public class GnPlaylist : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnPlaylist(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnPlaylist obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnPlaylist() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnPlaylist(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

  public GnPlaylist() : this(gnsdk_csharp_marshalPINVOKE.new_GnPlaylist(), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/** @internal Version @endinternal
*  Retrieves the Playlist SDK version string.
*  @return version string if successful
*  @return GNSDK_NULL if not successful
*  <p><b>Remarks:</b></p>
*  You can call this API at any time after successfully getting a GnManager instance. The returned
*   string is a constant. Do not attempt to modify or delete.
*  Example version string: 1.2.3.123 (Major.Minor.Improvement.Build)
*  Major: New functionality
*  Minor: New or changed features
*  Improvement: Improvements and fixes
*  Build: Internal build number
* @ingroup Music_Playlist_InitializationFunctions
*/
  public string Version {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylist_Version_get(swigCPtr) );
	} 

  }

/** @internal BuildDate @endinternal
*  Retrieves the Playlist SDK's build date string.
*  @return Note Build date string of the format: YYYY-MM-DD hh:mm UTC
*  <p><b>Remarks:</b></p>
*  You can call this API at any time after successfully getting a GnManager instance. The returned
*   string is a constant. Do not attempt to modify or delete.
*  Example build date string: 2008-02-12 00:41 UTC
* @ingroup Music_Playlist_InitializationFunctions
*/
  public string BuildDate {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnPlaylist_BuildDate_get(swigCPtr) );
	} 

  }

  public GnPlaylistAttributeEnumerable AttributesSupported {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnPlaylist_AttributesSupported_get(swigCPtr);
      GnPlaylistAttributeEnumerable ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnPlaylistAttributeEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnPlaylist.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b>: Provides access to information regarding the underlying 
* GNSDK playlist library. For working with user collections to create playlists 
* see {@link GnPlaylistCollection}. 
*/ 
 
public class GnPlaylist {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnPlaylist(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnPlaylist obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnPlaylist(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public GnPlaylist() throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnPlaylist(), true);
  }

/** 
*  Retrieves the Playlist SDK version string. 
*  @return version string if successful 
*  @return GNSDK_NULL if not successful 
*  <p><b>Remarks:</b></p> 
*  You can call this API at any time after successfully getting a {@link GnManager} instance. The returned 
*   string is a constant. Do not attempt to modify or delete. 
*  Example version string: 1.2.3.123 (Major.Minor.Improvement.Build) 
*  Major: New functionality 
*  Minor: New or changed features 
*  Improvement: Improvements and fixes 
*  Build: Internal build number 
* 
*/ 
 
  public static String version() {
    return gnsdk_javaJNI.GnPlaylist_version();
  }

/** 
*  Retrieves the Playlist SDK's build date string. 
*  @return Note Build date string of the format: YYYY-MM-DD hh:mm UTC 
*  <p><b>Remarks:</b></p> 
*  You can call this API at any time after successfully getting a {@link GnManager} instance. The returned 
*   string is a constant. Do not attempt to modify or delete. 
*  Example build date string: 2008-02-12 00:41 UTC 
* 
*/ 
 
  public static String buildDate() {
    return gnsdk_javaJNI.GnPlaylist_buildDate();
  }

/** 
*  Media attributes support for playlist generation such as album title, 
*  artist name, and artist type. 
*  @return Playlist attribute iterator 
*/ 
 
  public GnPlaylistAttributeIterable attributesSupported() {
    return new GnPlaylistAttributeIterable(gnsdk_javaJNI.GnPlaylist_attributesSupported(swigCPtr, this), true);
  }

}
